pyproject.toml

```diff
@@ -19,15 +19,19 @@ dependencies = [
 
 [project.optional-dependencies]
 jupyterlite = [
-    "jupyterlite-core[all]~=0.3.0",
-    "jupyterlite-pyodide-kernel~=0.3.2",
+    "jupyterlite-core[all]~=0.4.0",
+    "jupyterlite-pyodide-kernel~=0.4.1",
     "ipython~=8.26.0",
     # See https://github.com/ipython/ipython/issues/14303#issuecomment-1925423956
     "executing@git+https://github.com/alexmojaki/executing@3.13#egg=executing",
-    "sphinx-autobuild"
+    "sphinx-autobuild",
+    "attrs==23.2.0", # The version 24.2.0 supports only latest Python 3.14 codebase
```

Comment:
@davepeck 
The tag-strings branch doesn't merge the latest main.
And, the branch doesn't have annotationlib because it's too old.

Reply:
Ah, got it. We should update that! But for now, seems fine to merge this PR?